### ISSUE TITLE
How can i change the active state and run custom function?

### ISSUE THREAD

**Issue by marcelalburg** [react-bootstrap/react-bootstrap#1539]
Hello,

i using react bootstrap and it's great how easy it works. Now, i need to use a "onSelect" function (in my case) at Tabs and i don't know how i can set the correct active Tab after click.

1st Case)

```
changeTab(tab) {
}

render() {
  return (<Tabs onSelect={this.changeTab}>...</Tabs>)
}
```

the active state don't change automatically and i cannot call "setState" from "Tabs"

2nd Case)

before i set the active state, i will call a own function (i need to access the context of this component)

```
second() {
  this.context......
}

changeTab(tab) {
  this.second();
}

render() {
  return (<Tabs onSelect={this.changeTab.bind(this)}>...</Tabs>)
}
```

How can i set the active state of the Tabs in both cases ?

Thanks a lot
Marcel

**Comment by taion** [comment]
Follow this example: http://react-bootstrap.github.io/components.html#tabs-controlled

**Comment by marcelalburg** [comment]
Hello,

thanks for the really fast answer.

I see that and checked your code and figure out that can be easier. This approach is to much for me. I need to create a Component for every type of data, just to highlight the tabs.

i see in your code this (Tabs.js):

```
  handleSelect: function handleSelect(selectedKey) {
    if (this.props.onSelect) {
      this._isChanging = true;
      this.props.onSelect(selectedKey);
      this._isChanging = false;
      return;
    }

    // if there is no external handler, then use embedded one
    var previousActiveKey = this.getActiveKey();
    if (selectedKey !== previousActiveKey) {
      this.setState({
        activeKey: selectedKey,
        previousActiveKey: previousActiveKey
      });
    }
  },

```

From my Point of View, it's possible to change that to:

```
handleSelect: function handleSelect(selectedKey) {
    if (this.props.onSelect) {
      this._isChanging = true;
      if (this.props.onSelect(selectedKey) === true)
      {
        this._isChanging = false;
        return;
      }
        this._isChanging = false;
    }
...
```

everything should work again, and it's not necessary to create a own component for every type of tab content.

But maybe i'm wrong.

Thanks
Marcel

**Comment by taion** [comment]
Hmm - actually, I wonder if we'd be better off _only_ looking at the injected active key, sort of like how `<input>`s only look at `value`. @jquense What do you think?

**Comment by jquense** [comment]
#1109 kills this custom behavior anyway and just uses `uncontrollable` I believe. Not sure if that addresses your thought tho.

**Comment by taion** [comment]
Oops, thought we merged that. The thought is basically - if you pass in `onSelect` but not `activeKey`, you should still get an uncontrolled component. Do you know offhand if #1525 gives that behavior?

**Comment by jquense** [comment]
yeah, uncontrollable only checks for the existence of value prop not the handler. You should be able to listen for select without controlling it.

**Comment by taion** [comment]
Sweet, resolved by #1525 then.
